### PR TITLE
Run Tests in CI

### PR DESCRIPTION
**Pull request by charlie-tf**
Tests were not being run in CI—run them.

### CODE REVIEW

**Comment by charlie-tf** [comment]
Ugh, i had successfully run these locally until I ran `yarn` again, which upgraded `babel-loader`. Now I'm getting the same error locally as here.

**Comment by karthikbtf** [comment]
> Ugh, i had successfully run these locally until I ran `yarn` again, which upgraded `babel-loader`. Now I'm getting the same error locally as here. So, "yay," but also "ugh."

@charlie-tf you can try updating the version to 8.1.0 under package.json resolutions. Let me see what happens if we remove it (i think i added it for build to work).
I see from running locally that yarn test updates tsconfig.json - adds `noEmit: true`.. we need to remove the existing `emitDeclarationOnly: true` for build to succeed

File: package.json [hunk 1]
```diff
@@ -95,12 +95,12 @@
     "@storybook/react": "^6.3.7",
     "@svgr/cli": "^6.3.1",
     "@testing-library/jest-dom": "^5.11.4",
-    "@testing-library/react": "^13.0.0",
+    "@testing-library/react": "^12.1.5",
```
Comment (charlie-tf) [inline package.json:98]: was getting an error. changed per [stackoverflow advice](https://stackoverflow.com/a/72183519/20167578) which got things running again.

Comment (charlie-tf) [reply package.json:98]: ...which seems to have caused deploy to fail 😞

Comment (karthikbtf) [reply package.json]: What error were you seeing prior to applying the stackoverflow advice? 
I was able to build and run storybook after removing babel-loader.

Comment (charlie-tf) [reply package.json]: After removing the `babel-loader` resolution, I see the same error:
```
There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "babel-loader": "8.1.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of babel-loader was detected higher up in the tree:

  /Users/charliekeith/workspace/web-components/node_modules/babel-loader (version: 8.3.0) 
...
```
After upgrading `react-scripts`, I see:
```
    Cannot find module 'react-dom/client' from 'node_modules/@testing-library/react/dist/pure.js'

    Require stack:
      node_modules/@testing-library/react/dist/pure.js
      node_modules/@testing-library/react/dist/index.js
      src/components/Button/button.test.tsx

      at Resolver.resolveModule (node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (node_modules/@testing-library/react/dist/pure.js:35:46)
```
Downgrading `@testing-library/react` fixes that, but I guess it messes with enough other dependencies that the deploy failed.

Comment (karthikbtf) [reply package.json]: You probably needed to nuke node-modules and reinstall.. that worked for me.

Comment (karthikbtf) [reply package.json]: The 8.3.1 was most likely from a previous install

Comment (charlie-tf) [reply package.json]: did you try running `yarn test`?

Comment (karthikbtf) [reply package.json]: yes, it ran successfully.. let me try again.

Comment (karthikbtf) [reply package.json]: yes...
```
No tests found related to files changed since last commit.
Press `a` to run all tests, or run Jest with `--watchAll`.

Watch Usage
 › Press a to run all tests.
 › Press f to run only failed tests.
 › Press q to quit watch mode.
 › Press p to filter by a filename regex pattern.
 › Press t to filter by a test name regex pattern.
 › Press Enter to trigger a test run.

```

Comment (charlie-tf) [reply package.json]: huh, `rm`ing node_modules doesn't seem to help—I get the same error about babel-loader 8.3.0... This is what `npm ls babel-loader` gives me. Does it look similar to yours?
```
charliekeith:web-components$ npm ls babel-loader
@terraware/web-components@2.0.56 /Users/charliekeith/workspace/web-components
├─┬ @storybook/addon-essentials@6.5.15
│ ├─┬ @storybook/addon-docs@6.5.15
│ │ └── babel-loader@8.3.0
│ └─┬ @storybook/core-common@6.5.15
│   └── babel-loader@8.3.0 deduped
├─┬ @storybook/react@6.5.15
│ └─┬ @storybook/core@6.5.15
│   └─┬ @storybook/core-server@6.5.15
│     ├─┬ @storybook/builder-webpack4@6.5.15
│     │ └── babel-loader@8.3.0 deduped
│     └─┬ @storybook/manager-webpack4@6.5.15
│       └── babel-loader@8.3.0 deduped
└─┬ react-scripts@4.0.3
  └── babel-loader@8.1.0
```

Comment (karthikbtf) [reply package.json]: > huh, `rm`ing node_modules doesn't seem to help—I get the same error about babel-loader 8.3.0... This is what `npm ls babel-loader` gives me. Does it look similar to yours?
> 
> ```
> charliekeith:web-components$ npm ls babel-loader
> @terraware/web-components@2.0.56 /Users/charliekeith/workspace/web-components
> ├─┬ @storybook/addon-essentials@6.5.15
> │ ├─┬ @storybook/addon-docs@6.5.15
> │ │ └── babel-loader@8.3.0
> │ └─┬ @storybook/core-common@6.5.15
> │   └── babel-loader@8.3.0 deduped
> ├─┬ @storybook/react@6.5.15
> │ └─┬ @storybook/core@6.5.15
> │   └─┬ @storybook/core-server@6.5.15
> │     ├─┬ @storybook/builder-webpack4@6.5.15
> │     │ └── babel-loader@8.3.0 deduped
> │     └─┬ @storybook/manager-webpack4@6.5.15
> │       └── babel-loader@8.3.0 deduped
> └─┬ react-scripts@4.0.3
>   └── babel-loader@8.1.0
> ```

```
=> Found "babel-loader@8.3.0"
info Reasons this module exists
   - "react-scripts" depends on it
   - Hoisted from "react-scripts#babel-loader"
   - Hoisted from "@storybook#react#@storybook#core-common#babel-loader"
   - Hoisted from "@storybook#addon-essentials#@storybook#addon-docs#@storybook#builder-webpack4#babel-loader"
   - Hoisted from "@storybook#react#@storybook#core#@storybook#core-server#@storybook#manager-webpack4#babel-loader"
info Disk size without dependencies: "576KB"
info Disk size with unique dependencies: "884KB"
info Disk size with transitive dependencies: "3.93MB"
info Number of shared dependencies: 24
✨  Done in 0.38s.
MacBook-Pro:web-components kb$ npm ls babel-loader
@terraware/web-components@2.0.56 /Users/kb/Work/terraformation/web-components
├─┬ @storybook/addon-essentials@6.3.7
│ ├─┬ @storybook/addon-docs@6.3.7
│ │ └─┬ @storybook/builder-webpack4@6.3.7
│ │   └── babel-loader@8.3.0
│ └── babel-loader@8.1.0
├─┬ @storybook/react@6.3.7
│ ├─┬ @storybook/core-common@6.3.7
│ │ └── babel-loader@8.3.0
│ └─┬ @storybook/core@6.3.7
│   └─┬ @storybook/core-server@6.3.7
│     └─┬ @storybook/manager-webpack4@6.3.7
│       └── babel-loader@8.3.0
└─┬ react-scripts@4.0.3
  └── babel-loader@8.1.0 deduped
```
oh i forgot, i ran `yarn add babel-loader@8.1.0` which updated my yarn.lock to 8.1.0... and i removed it from package.json - this was part of some test.. probably why it's working for me. we don't have the same environments.

I think this should work, add babel-loader back to resolutions in package.json - but set the version to 8.1.0.
react shouldn't complain after that.

Comment (karthikbtf) [reply package.json]: this worked for me "add babel-loader back to resolutions in package.json - but set the version to 8.1.0." 
build, storybook test.. all of 'em
additional change was removing "emitDeclarationOnly" setting from tsconfig.json

**Review by karthikbtf — APPROVED** [review]
thanks for doing this
